library/data/registry_entry_attributes/src/main/java/org/quiltmc/qsl/registry/attribute/api/RegistryExtensions.java

```diff
@@ -0,0 +1,65 @@
+/*
+ * Copyright 2021 QuiltMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.quiltmc.qsl.registry.attribute.api;
+
+import java.util.function.Consumer;
+
+import net.minecraft.util.Identifier;
+import net.minecraft.util.registry.Registry;
+
+import org.quiltmc.qsl.registry.attribute.impl.DefaultAttributesBuilderImpl;
+
+/**
+ * Extensions for working with {@link Registry}s.
+ */
+public final class RegistryExtensions {
+	/**
+	 * Registers an entry into a registry, also setting built-in attribute values for it.
+	 *
+	 * @param registry        target registry
+	 * @param id              entry identifier
+	 * @param entry           entry
```

Comment:
Might want a better description on this - maybe call it `toRegister` or something? It's not super clear that This Is What You Are Registering from the name/docs